clients/githubrepo/client.go

```diff
@@ -177,6 +177,24 @@ func (client *Client) ListStatuses(ref string) ([]clients.Status, error) {
 	return client.statuses.listStatuses(ref)
 }
 
+//ListProgrammingLanguages implements RepoClient.ListProgrammingLanguages.
+func (client *Client) ListProgrammingLanguages() (map[string]int, error) {
```

Comment:
we need to cache these results and not issue the query on each API call to `ListProgrammingLanguages`. Can you look into adding support for caching? (You can look into other APIs, we already do it). Basically in this function we should only do:
```
return client._some_function_();
```

Reply:
np, I'll look into it

Reply:
@laurentsimon please review my implementation for caching the list langs at https://github.com/ossf/scorecard/pull/1979/files#diff-31e1ee0c52fdbfa08b1f646de83cdae712e6a9fbe98c0deabb38fa36e533c04bR184 and https://github.com/ossf/scorecard/pull/1979/files#diff-8214bff6f7acf7713c122c9b398050b1464aca5ec3153a03d8ebf16bf0c95f81R36.

Reply:
follow the webhook example https://github.com/aidenwang9867/scorecard/blob/a9180258610da4800e4798df6efaa6d47bc5ee48/clients/githubrepo/webhook.go#L28

The API call should happen once only, then you return the cached data